service/src/test/java/org/broadinstitute/listener/relay/http/RelayedHttpRequestProcessorTest.java

```diff
@@ -241,16 +247,20 @@ void writeTargetResponseOnCaller_setsNoSniff() throws IOException {
 
   @Test
   void writeTargetResponseOnCaller_withBodyResponseStreamsClose() throws IOException {
+    // this test uses a spy to verify we close its stream
+    InputStream spyBody =
+        Mockito.spy(new ByteArrayInputStream(BODY_CONTENT.getBytes(StandardCharsets.UTF_8)));
+
     when(targetHttpResponse.getContext()).thenReturn(context);
-    when(targetHttpResponse.getBody()).thenReturn(Optional.of(mockBody));
+    when(targetHttpResponse.getBody()).thenReturn(Optional.of(spyBody));
     when(targetHttpResponse.getStatusCode()).thenReturn(200);
     when(context.getResponse()).thenReturn(listenerResponse);
     when(targetHttpResponse.getCallerResponseOutputStream()).thenReturn(responseStream);
 
     processor.writeTargetResponseOnCaller(targetHttpResponse);
 
     verify(responseStream).close();
-    verify(mockBody).close();
+    verify(spyBody).close();
```

Comment:
I'm not sure what was going on with this test. It previous used `mockBody`, which was a `@Mock InputStream` … but at no point was that mock set up to contain any content or have its `.read()` etc. methods give any responses. This is also the only test that used `mockBody`.

So, I changed it to use a spy with some actual content, and made the spy local to the test.